### ISSUE TITLE
abi::__cxa_demangle

### ISSUE THREAD

**Issue by nickjbrowning** [lab-cosmo/sphericart#153]
> OK, could we just remove the depencency on the GCC-specific `abi::__cxa_demangle`? And use the mangled type here instead?

_Originally posted by @Luthaf in https://github.com/lab-cosmo/sphericart/pull/131#discussion_r1799609138_

**Comment by nickjbrowning** [comment]
No, because we need the unmangled types to construct e.g my_kernel<float> or my_kernel<double, int>. See here:

https://github.com/lab-cosmo/sphericart/blob/028b2742dabf6511e17e18f5c377f89f359da4a5/sphericart/src/cuda_base.cpp#L112

Do you have a better idea of doing this? The way I'm doing it is exactly the same way that NVRTC does it in their backend API...

**Comment by Luthaf** [comment]
Hmm, alright. Then it would be nice to move this code into a `demangle_symbol` function, with the appropriate `#ifdef` (like for `dlopen` & co). In the current state, I am a bit worried that we are making it a lot harder to get a windows build down the line.

**Comment by nickjbrowning** [comment]
windows has an equivalent function thats very similar so not too concerning. We can create a wrapper and ifdef the calls no problem.

**Comment by Luthaf** [comment]
I saw that the calls are already in an ifdef, so this should be doable later down the line. I would still create a wrapper to handle platform differences, but this should be good for now!
